src/irmin-pack/inode.ml

```diff
@@ -1055,31 +1171,31 @@ struct
       | Unknown | Static _ -> assert false
       | Dynamic f -> f
 
-    let encode_bin ~dict ~offset (t : t) k =
+    let encode_bin ~dict ~offset_of_key hash (t : t) =
       Stats.incr_inode_encode_bin ();
       let step s : Compress.name =
         let str = step_to_bin s in
         if String.length str <= 3 then Direct s
         else match dict str with Some i -> Indirect i | None -> Direct s
       in
-      let hash h : Compress.address =
-        match offset h with
-        | None -> Compress.Direct h
+      let address_of_key key : Compress.address =
+        match offset_of_key key with
+        | None -> Compress.Direct hash
```

Comment:
This introduced a bug!

Should be `(Key.to_hash key)` here... `hash` is the hash of the value _being encoded_.

Reply:
😨